### PR TITLE
[CBRD-20163] Log old overflow versions resulted from updates

### PR DESCRIPTION
**Pull request by andrei14vl**
http://jira.cubrid.org/browse/CBRD-20163

See also discussion on #234 .

Old overflow records will be kept in log, in the same way other type of records are handled.

Vacuum role is reduced and accessing old records is simplified. 

One important point is that the resulted log record will be used only to access the visible version of the record and it will not have any role in the actual undo. Overflow undo is still done by logging each overflow page and undoing them independently.

### CODE REVIEW

File: src/storage/heap_file.c [hunk 1]
```diff
@@ -21840,7 +21821,7 @@ heap_update_relocation (THREAD_ENTRY * thread_p, HEAP_OPERATION_CONTEXT * contex
 	  p_addr.offset = context->oid.slotid;
 
 	  /* home remains untouched, log no_change on home to notify vacuum */
-	  heap_mvcc_log_home_no_change_on_delete (thread_p, &p_addr);
+	  heap_mvcc_log_home_no_change (thread_p, &p_addr);
```
Comment (ribram) [inline src/storage/heap_file.c]: I think we have a bug in the code:
```c
      heap_log_update_physical (thread_p, context->forward_page_watcher_p->pgptr, &context->hfid.vfid, &forward_oid,
				&forward_recdes, context->recdes_p,
				(is_mvcc_op ? RVHF_UPDATE_NOTIFY_VACUUM : RVHF_UPDATE));
```

We don't need to notify vacuum of forward record update. We notify it on home page. Should be changed to:
```c
      heap_log_update_physical (thread_p, context->forward_page_watcher_p->pgptr, &context->hfid.vfid, &forward_oid,
				&forward_recdes, context->recdes_p, RVHF_UPDATE);
```

File: src/transaction/recovery.c [hunk 2]
```diff
@@ -972,10 +972,17 @@ struct rvfun RV_fun[] = {
 
   {RVHF_MVCC_UPDATE_OVERFLOW,
    "RVHF_MVCC_UPDATE_OVERFLOW",
-   heap_rv_undo_update,
-   heap_rv_redo_update_and_update_chain,
-   log_rv_dump_hexa,
-   log_rv_dump_hexa}
+   heap_rv_nop,
+   heap_rv_nop,
```
Comment (ribram) [inline src/transaction/recovery.c]: For redo, I think instead heap_rv_nop we need heap_rv_update_chain_after_mvcc_op.

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/storage/overflow_file.c (1 incorrect indentation(s) found)
```
  1426 }
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/transaction/log_manager.c (1 incorrect indentation(s) found)
```
  9875 	{
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by eseokoh** [comment]
Please find Alex's change from https://github.com/CUBRID/cubrid/pull/273

**Comment by eseokoh** [comment]
@andrei14vl 

Please let me remind you can merge the fix to let QA follow you. :)

**Pull request merged**